### PR TITLE
Set QT_GLOBAL_SCALE for better HiDPI support

### PR DESCRIPTION
**Pull request by waddlesplash**
Not perfect, but definitely better than before.

No scaling (12pt, 1x):
image

Before this change, 18pt (1.5x):
image

After (same):
image

### CODE REVIEW

File: platforms/qhaikuplatform/haiku/qhaikuwindow.cpp [hunk 1]
```diff
@@ -915,7 +916,8 @@ void QHaikuWindow::swapBuffers()
 
 BRegion QHaikuWindow::getClippingRegion()
 {
-	BRegion region(BRect(0, 0, window()->width(), window()->height()));
+	const QSize nativeSize = QHighDpi::toNativePixels(window()->size(), window());
```
Comment (ilya-fedin) [inline platforms/qhaikuplatform/haiku/qhaikuwindow.cpp]:
```suggestion
	const QSize nativeSize = windowGeometry().size();
```

See QPlatformWindow::windowGeometry()

File: platforms/qhaikuplatform/haiku/qhaikuscreen.cpp [hunk 2]
```diff
@@ -58,6 +59,9 @@ QHaikuScreen::QHaikuScreen()
 	, m_screen(new BScreen(B_MAIN_SCREEN_ID))
 {
 	Q_ASSERT(m_screen->IsValid());
+
+	// Set a global scale factor instead of a per-screen one (for now.)
+	QHighDpiScaling::setGlobalFactor(std::max(1.0f, be_plain_font->Size() / 12.0f));
```
Comment (ilya-fedin) [inline platforms/qhaikuplatform/haiku/qhaikuscreen.cpp]: This is likely to break various applications as this force overrides the scale factor, it won't respect the [high-DPI scale factor rounding policy](https://doc.qt.io/qt-5/qguiapplication.html#highDpiScaleFactorRoundingPolicy) set by application anymore. You should implement QHaikuScreen::logicalDpi and QHaikuScreen::logicalBaseDpi instead. You can send DPI change event with QWindowSystemInterface::handleScreenLogicalDotsPerInchChange.

Comment (waddlesplash) [reply platforms/qhaikuplatform/haiku/qhaikuscreen.cpp]: I implemented those two APIs originally and it did not seem to work correctly. It wasn't just that the correct scale factor wasn't used, it was that different things (icons, text, etc.) were scaled with different factors when I tried that. Even when I returned values that were integer multiples of each other, I still did not get the nice results that this provides. What should I do about that?

Comment (ilya-fedin) [reply platforms/qhaikuplatform/haiku/qhaikuscreen.cpp]: Qt 5 applications need to manually activate High-DPI scaling support, apparently the ones you tried don't do that. Qt 6 has High-DPI support always active. `QT_ENABLE_HIGHDPI_SCALING=1` should override that unless the application explicitly disables High-DPI scaling support (available only in Qt 5 API though).

File: platforms/qhaikuplatform/haiku/qhaikuwindow.cpp (hunk 1 above)
Comment (waddlesplash) [reply platforms/qhaikuplatform/haiku/qhaikuwindow.cpp]: Aha, indeed.

**Comment by waddlesplash** [comment]
I now get the same behavior in the screenshots above only when applying `QT_SCALE_FACTOR_ROUNDING_POLICY=PassThrough QT_ENABLE_HIGHDPI_SCALING=1` but I guess that's how it should be.

**Comment by ilya-fedin** [comment]
Yeah, Qt 6 has such behavior out of the box though

**Comment by waddlesplash** [comment]
@threedeyes, @diversys: this has been ready to merge for a while, can it be merged and a new version pushed?

**Pull request merged**